config/config.go

```diff
@@ -66,6 +67,10 @@ func LoadFile(filename string) (*Config, error) {
 
 // The defaults applied before parsing the respective config sections.
 var (
+	// CurrentConfig is a pointer to the current configuration.
```

Comment:
A global isn't how we pass config around, see https://github.com/prometheus/prometheus/blob/master/cmd/prometheus/main.go#L401

Reply:
I know.